include/dali/core/mm/pool_resource.h

```diff
@@ -349,21 +258,15 @@ class pool_resource_base : public memory_resource<kind, Context> {
     size_t blk_size = bytes;
     void *new_block = get_upstream_block(blk_size, bytes, alignment);
     assert(new_block);
-    try {
+    if (blk_size == bytes) {
```

Comment:
Why you removed try-catch?

Reply:
What should be guarded with try/catch is the insertion into `blocks_` - that has been moved to `get_upstream_block` (which now uses a lock and therefore it's safe to do it there). The code was incorrect, because if there was an exception _after_ inserting into `blocks_` (e.g. while inserting into `free_list`), then we'd have a dead block in `blocks_` and that would result in double free when freeing to the upstream.

Reply:
Ok